packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -297,32 +352,85 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
     }
   };
 
-  const handleFirstChars = (id, firstChar) => {
+  const mapFirstChar = (id, firstChar) => {
     firstCharMap.current[id] = firstChar;
   };
 
+  const prevChildIds = React.useRef([]);
+  const [mounted, setMounted] = React.useState(false);
+  React.useEffect(() => {
+    const childIds = React.Children.map(children, child => child.props.nodeId) || [];
+    if (arrayDiff(prevChildIds.current, childIds)) {
+      nodeMap.current[-1] = { parent: null, children: childIds };
+
+      childIds.forEach((id, index) => {
+        if (index === 0) {
+          setTabbable(id);
+        }
+        nodeMap.current[id] = { parent: null };
+      });
+      visibleNodes.current = nodeMap.current[-1].children;
+      prevChildIds.current = childIds;
+      setMounted(true);
```

Comment:
This conditionally sets mounted even though "mounted" is not something that is conditional.

This should live in it's own effect, preferable hook `useMountedState`. And we should document why we need to re-render after the first commit.

Reply:
I guess the name would be wrong here - this doesn't specifically check for the state of the component only that the effect has run